services/security-headers/security-headers.service.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const { BaseService } = require('..')
+
+const documentation = `
+<p>
+  The <a href="https://securityheaders.com/">Security Headers</a>
+  provide an easy mechanism to analyze HTTP response headers and
+  give information on how to deploy missing headers.
+</p>
+</p>
+  The scan result will be hidden from the public result list and follow redirects will be on too.
+<p>
+`
+
+module.exports = class SecurityHeaders extends BaseService {
+  static get category() {
+    return 'monitoring'
+  }
+
+  static get route() {
+    return {
+      base: 'security-headers',
+      pattern: ':protocol(https|http)/:host',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Security Headers',
+        namedParams: {
+          protocol: 'https',
+          host: 'shields.io',
+        },
+        staticPreview: this.render({
+          grade: 'A+',
+        }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'security headers',
+    }
+  }
+
+  static render({ grade }) {
+    const colorMap = {
+      'A+': 'brightgreen',
+      A: 'green',
+      B: 'yellow',
+      C: 'yellow',
+      D: 'orange',
+      E: 'orange',
+      F: 'red',
+    }
+
+    return {
+      message: grade,
+      color: colorMap[grade],
+    }
+  }
+
+  async handle({ protocol, host }) {
+    const { res } = await this._request({
+      url: `https://securityheaders.com`,
+      options: {
+        method: 'HEAD',
+        qs: {
+          q: `${protocol}://${host}`,
+          hide: 'on',
+          followRedirects: 'on',
+        },
+      },
+    })
+
+    const grade = res.headers['x-grade']
```

Comment:
Is it worth checking whether grade is defined? Something along the lines of:
````
if (!grade) {
  throw new NotFound({ prettyMessage: 'not available' }
}

````

Reply:
I agree. I added the suggested code.